packages/base64/test/bench-main.js

```diff
@@ -0,0 +1,46 @@
+import { encodeBase64, decodeBase64 } from '../src/main';
+
+const string = new Array(100)
+  .fill(
+    'there once a rich man from nottingham who tried to cross the river. what a dope, he tripped on a rope. now look at him shiver.',
+  )
+  .join(' ');
+const data = encodeBase64(string);
+
+{
+  const start = Date.now();
+  const limit = start + 1000;
+  let operations = 0;
+  for (let n = 1; Date.now() < limit; n *= 2) {
+    for (let i = 0; i < n; i += 1) {
+      encodeBase64(string);
+    }
+    operations += n;
+  }
+  const end = Date.now();
+  const duration = end - start;
+  console.log(
+    operations / duration,
+    'encodes per millisecond, for input of length',
+    string.length,
+  );
+}
```

Comment:
I have not been paying attention to our benchmarking code, but this is a nice pattern I had not seem before. However, only line 16 is not boilerplate. Can we package the rest up as a reusable abstraction? And then reuse it below? Would the other places we benchmark be able to reuse it as well?

Reply:
I can likely factor out a benchmarking abstraction that does everything in its power to avoid capturing wrapper overhead in its results.

@warner Is there a bench tool that we’re using elsewhere?

Reply:
Given above answer, resolving.